contrib/dockerd-rootless-setuptool.sh

```diff
@@ -239,11 +239,11 @@ init() {
 
 	# instructions: validate subuid for current user
 	if command -v "getsubids" > /dev/null 2>&1; then
-		getsubids "$USERNAME" > /dev/null 2>&1 || getsubids "$(id -u)" > /dev/null 2>&1
+		testcommand="getsubids '$USERNAME' > /dev/null 2>&1 || getsubids '$(id -u)' > /dev/null 2>&1"
 	else
-		grep -q "^$USERNAME_ESCAPED:\|^$(id -u):" /etc/subuid 2> /dev/null
+		testcommand="grep -q '^$USERNAME_ESCAPED:\|^$(id -u):' /etc/subuid 2> /dev/null"
 	fi
-	if [ $? -ne 0 ]; then
+	if ! eval "$testcommand"; then
```

Comment:
`eval` can be a source of vulnerability.

Instead of this, a script could be like
```sh
error_subid=
if ...; then
  getsubids ... || error_subid=1
else
  grep ... || error_subid=1
fi

if [ "$error_subid" = "1" ] ; then
...
fi
```

Reply:
Nice. Incorporated the suggestion in my new commit.